core/arch/arm/kernel/boot.c

```diff
@@ -1296,13 +1308,19 @@ static void init_primary(unsigned long pageable_part, unsigned long nsec_entry)
  * Note: this function is weak just to make it possible to exclude it from
  * the unpaged area.
  */
-void __weak boot_init_primary_late(unsigned long fdt)
+void __weak boot_init_primary_late(unsigned long __maybe_unused fdt)
 {
+#ifdef CFG_EXTERNAL_DT
 	init_external_dt(fdt);
 	tpm_map_log_area(get_external_dt());
 	discover_nsec_memory();
 	update_external_dt();
 	configure_console_from_dt();
+#else
+	tpm_map_log_area(get_embedded_dt());
```

Comment:
This is not applicable. TPM log buffer info cannot be found from the embedded DT.
Note `tpm_map_log_area(NULL);` will not harm the system but only emit info trace messages.